src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the storage for user's task list.
+ * Reads and writes data into user's saved file that stores task list.
+ */
+public class Storage {
+    private final String filePath;
+    private final Parser parser = new Parser();
+
+    /**
+     * Constructor for Storage object.
+     * Makes file tasks.txt if non-existent. Else, validates existence of file
+     *
+     * @param filePath
+     * @throws IOException if filePath is not "data/tasks.txt"
+     * @see Files
+     * @see Paths
+     */
+    public Storage(String filePath) throws IOException {
+        String parentPath = filePath.split("tasks.txt")[0];
+        if (!Files.exists(Paths.get(parentPath))) {
+            Files.createDirectories(Paths.get(parentPath));
+        }
+        if (!Files.exists(Paths.get(filePath))) {
+            Files.createFile(Paths.get(filePath));
+        }
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns user's task list that is in the saved file on user's local disk.
+     * Initializes objects using the String data in tasks.txt file.
+     * Parses the data saved and adds correct form of tasks in the TaskList object
+     *
+     * @return ArrayList of user's tasks.
+     * @throws IOException if incorrect data in file or incorrectly parsed.
+     * @see Parser
+     * @see TaskList
+     * @see Task
+     * @see ToDo
+     * @see Event
+     * @see Deadline
+     * @see Paths
+     */
+    public ArrayList<Task> getTaskList() throws IOException {
+        ArrayList<Task> taskList = new ArrayList<>();
+        List<String> fileLines = Files.readAllLines(Paths.get(filePath));
+        for (int i = 0; i < fileLines.size(); i++) {
+            String line = fileLines.get(i);
+            String[] userTask = line.split(" \\| ");
```

Comment:
Distinguish clearly between single-valued and multi-valued variables. Use plural instead

Reply:
To add on what internityz has mentioned above, for the line 53 code in Storage.java, you can consider renaming the Arraylist<Task> taskList to tasks.